starwars-characters/src/App.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import Character from './character'
+import './App.css';
+import HomeWorld from './homeworld'; 
+import CharacterList from './characters';
+import { BrowserRouter as Router, Route, Link} from 'react-router-dom';
+
+function Home(){
+return <div><CharacterList /></div>
+}
+
+function Char(){
+  return <div>{<Character />}</div>
+}
+
+function hWorld(){
+  return <div>{<HomeWorld />}</div>
+}
+
+class App extends React.Component {
+  
+  render(){
+    
+    return (
+      <div>
+        <Router>
+          <Link to='/' >Skywalker Family</Link><br />
+          <Link to="/hworld">HomeWorld</Link><br />
+          <Link to="/char">Character Details</Link>
+          <hr />
+          <switch>
+            <Route exact path='/' component={Home} />
+            <Route path='/char' component={Char}/>
```

Comment:
you need to add a link param to catch whatever id is passed after 'char/' that way you can use it decide what to fetch.
`<Route path='/char/:id' component={Character}/>`
Also see my note about getting rid of the function components.